src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/delete_interaction_choice_set_request.cc

```diff
@@ -157,10 +185,46 @@ void DeleteInteractionChoiceSetRequest::SendVrDeleteCommand(
   msg_params[strings::type] = hmi_apis::Common_VRCommandType::Choice;
   msg_params[strings::grammar_id] = (*choice_set)[strings::grammar_id];
   choice_set = &((*choice_set)[strings::choice_set]);
-  for (uint32_t i = 0; i < (*choice_set).length(); ++i) {
-    msg_params[strings::cmd_id] = (*choice_set)[i][strings::choice_id];
-    SendHMIRequest(hmi_apis::FunctionID::VR_DeleteCommand, &msg_params);
+  {
+    sync_primitives::AutoLock auto_lock(requests_lock_);
+    for (uint32_t i = 0; i < (*choice_set).length(); ++i) {
+      msg_params[strings::cmd_id] = (*choice_set)[i][strings::choice_id];
+      const uint32_t delte_cmd_hmi_corr_id = SendHMIRequest(
+          hmi_apis::FunctionID::VR_DeleteCommand, &msg_params, true);
+      sent_requests_.insert(delte_cmd_hmi_corr_id);
+    }
+  }
+}
+
+void DeleteInteractionChoiceSetRequest::
+    SendDeleteInteractionChoiceSetResponse() {
+  const mobile_apis::Result::eType result_code =
+      MessageHelper::HMIToMobileResult(response_result_code_);
+  response_result_ =
+      helpers::Compare<mobile_apis::Result::eType, helpers::EQ, helpers::ONE>(
+          result_code,
+          mobile_apis::Result::SUCCESS,
+          mobile_apis::Result::WARNINGS);
+  if (response_result_) {
+    ApplicationSharedPtr app =
+        application_manager_.application(connection_key());
+    if (!app) {
+      LOG4CXX_ERROR(logger_,
+                    "Application with connection key " << connection_key()
+                                                       << " did not find.");
+      return;
+    }
+    const uint32_t choice_set_id =
+        (*message_)[strings::msg_params][strings::interaction_choice_set_id]
+            .asUInt();
+    app->RemoveChoiceSet(choice_set_id);
   }
+
+  LOG4CXX_DEBUG(
+      logger_,
+      "Response sent. Sussess: " << (response_result_ ? "true" : "false")
```

Comment:
@IGapchuk you can use `std::boolalpha` instead of `(response_result_ ? "true" : "false")`

Reply:
@BSolonenko Added in 05a0ac563f1b887f483b99518f63582d86643583.